src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -413,24 +414,121 @@ private static string EmitIndexOfAnyValues(char[] asciiChars, Dictionary<string,
+            for (int i = 0; i <= 0x7f; i++)
+            {
+                if (!RegexCharClass.CharInClass((char)i, set))
+                {
+                    asciiChars.Add((char)i);
+                }
+            }
+
+            string fieldName = EmitIndexOfAnyValues(asciiChars.ToArray(), requiredHelpers);
+            string? helperName = set switch
+            {
+                RegexCharClass.DigitClass => "IndexOfAnyDigit",
+                RegexCharClass.ControlClass => "IndexOfAnyControl",
+                RegexCharClass.LetterClass => "IndexOfAnyLetter",
+                RegexCharClass.LetterOrDigitClass => "IndexOfAnyLetterOrDigit",
+                RegexCharClass.LowerClass => "IndexOfAnyLower",
+                RegexCharClass.NumberClass => "IndexOfAnyNumber",
+                RegexCharClass.PunctuationClass => "IndexOfAnyPunctuation",
+                RegexCharClass.SeparatorClass => "IndexOfAnySeparator",
+                RegexCharClass.SpaceClass => "IndexOfAnyWhiteSpace",
+                RegexCharClass.SymbolClass => "IndexOfAnySymbol",
+                RegexCharClass.UpperClass => "IndexOfAnyUpper",
+                RegexCharClass.WordClass => "IndexOfAnyWordChar",
+                _ => null,
+            };
+            if (helperName is null)
+            {
+                using (SHA256 sha = SHA256.Create())
+                {
+                    helperName = $"IndexOfNonAsciiOrAny_{BitConverter.ToString(sha.ComputeHash(Encoding.UTF8.GetBytes(set))).Replace("-", "")}";
+                }
+            }
+
+            if (!requiredHelpers.ContainsKey(helperName))
+            {
+                var additionalDeclarations = new HashSet<string>();
+
+                var lines = new List<string>()
+                {
+                    $"/// <summary>Finds the next index of any character that matches {EscapeXmlComment(DescribeSet(set))}.</summary>",
+                    $"[MethodImpl(MethodImplOptions.AggressiveInlining)]",
+                    $"internal static int {helperName}(this ReadOnlySpan<char> span)",
+                    $"{{",
+                    $"    int i = span.IndexOfAnyExcept({fieldName});",
+                    $"    if ((uint)i < (uint)span.Length)",
+                    $"    {{",
+                    $"        if (span[i] <= 0x7f)",
+                    $"        {{",
+                    $"            return i;",
+                    $"        }}",
+                    $"",
+                    $"        do",
+                    $"        {{",
+                    $"            if ({MatchCharacterClass("span[i]", set, negate: false, additionalDeclarations, requiredHelpers)})",
+                    $"            {{",
+                    $"                return i;",
+                    $"            }}",
+                    $"            i++;",
+                    $"        }}",
+                    $"        while ((uint)i < (uint)span.Length);",
```

Comment:
Nit: Similar to #83349, this won't get rid of the bounds check here, but something like the following would
```c#
while (true)
{
    if ((uint)i >= (uint)span.Length) break;

    if (MatchCharacterClass(span[i]))
    {
        return i;
    }

    i++;
}
```

Reply:
Yeah.  I debated that.  Every time I work around this, though, I decrease the pressure on @egorbo :-)